cuqi/geometry/_geometry.py

```diff
@@ -702,32 +702,80 @@ def _compute_eigpairs(self, xnod, C_nu, sigma, M, N_GL):
 
 class StepExpansion(Continuous1D):
     '''
-    class representation of the step random field with 3 intermidiate steps
-    '''
-    def __init__(self, grid, axis_labels=['x'], **kwargs):
+    Class representation of step functions (piecewise constant functions) with `n_steps` 
+    equidistant steps on the interval [x0, xn], of length L.
+    The function `par2fun` maps the parameters `p` (which are the step magnitudes) to the 
+    corresponding step function evaluated on the spatial grid (`grid`) of nodes x0, x1, ...xn.
+    
+    For example, if `n_steps` is 3 and `grid` is a uniform grid on [0, L] with nodes x0=0, x1=0.1L, ..., xn=L, 
+    then the resulting function evaluated on the grid will be p[0], if x<=L/3, p[1], if L/3<x<=2L/3, 
+    p[2], if 2L/3<x<=L.
+    
+    Parameters:
+    -----------
+    grid: ndarray
+        | Regular grid points for the step expansion to be evaluated at. The number of grid points should be equal to or larger than `n_steps`. The latter setting can be useful, for example, when using the StepExpansion geometry as a domain geometry for a cuqipy :class:`Model` that expects the input to be interpolated on a (possibly fine) grid (`grid`). The interval endpoints, [x0, xn], should be included in the grid.
 
-        super().__init__(grid, axis_labels,**kwargs)
+    n_steps: int
+        | Number of equidistant steps.
 
-        L = self.grid[-1]
-        self._idx1 = np.where( (self.grid>=0*L)&(self.grid<=0.333*L) )
-        self._idx2 = np.where( (self.grid>0.333*L)&(self.grid<=0.666*L) )
-        self._idx3 = np.where( (self.grid>0.666*L)&(self.grid<=L) )
+    projection: str
```

Comment:
Perhaps clearer if we called this e.g. `fun2par_projection`?

Reply:
great just did that